x-pack/plugins/fleet/public/applications/fleet/sections/agents/index.tsx

```diff
@@ -31,6 +39,22 @@ export const FleetApp: React.FunctionComponent = () => {
 
   const fleetStatus = useFleetStatus();
 
+  const settings = useGetSettings();
+
+  const { cloud } = useStartServices();
```

Comment:
personally I would move this to inside `fleet_server_upgrade_modal.tsx` to encapsulate the logic there instead of passing it as a parameter